W-16376572-pt-4-alt-text-fa (#547)
mulesoft/docs-studio: modules/ROOT/pages/datamapper-visual-reference.adoc

.. Define the *Type* for your output data.

To define the type for your input data, click the edit icon

> image:edit-icon.jpg["A pencil on a notepad icon"] or the drop-down menu to the right of the *Type* input field. DataMapper displays a mapping configuration view according to the input type you selected. The image below displays the configuration view for input type `CSV`.

Reviewer: ```suggestion
image:edit-icon.jpg["A pencil on a notepad icon"] or the dropdown menu next to the *Type* input field. DataMapper displays a mapping configuration view according to the input type you selected. The following image displays the configuration view for input type `CSV`.
```

|1 |Display name input field. Defines the name for this DataMapper in the application flow as shown in the Studio canvas.
|2 |File name input field. Defines the name of the file that contains the data mapping.
|3 |Input source drop-down menu. By default, DataMapper takes its input from the message payload. Click to select another input source; you can select from several message properties or flow variables.

> |4 |Input type input field. Click the edit icon image:edit-icon.jpg["A pencil on a notepad icon"] or the drop-down menu to the right of the input field to select the input type. This causes the input pane to change allowing you to select between the available types for input data.

Reviewer: ```suggestion
|4 |Input type input field. Click the edit icon image:edit-icon.jpg["A pencil on a notepad icon"] or the dropdown menu next to the *Input Type* field. This causes the input pane to change allowing you to select between the available types for input data.
```

> |4 |Input type input field. Click the edit icon image:edit-icon.jpg["A pencil on a notepad icon"] or the drop-down menu to the right of the input field to select the input type. This causes the input pane to change allowing you to select between the available types for input data.

Reviewer: ```suggestion
|4 |Input type input field. Click the edit icon image:edit-icon.jpg["A pencil on a notepad icon"] or the dropdown menu next to the *Input Type* field. This causes the input pane to change allowing you to select between the available types for input data.
```

|5 |Output target input field. The default is the message payload. Select between payload or available properties, or write your own expression.
|6 |Output type drop-down menu. Click to select between the available types for output data.
|7 |Inbound/outbound properties view selector, available when the metadata view is enabled (see below).
|8 a|
Sidepanel view selector and config icons. Use these icons to:

> * image:view-selector-icon.png[A table icon with several rows and columns] / image:view-selector-icon-2.png["A green logo with a gold plus sign"] Toggle between the mapping view and the DataMapper configuration view (described below)

Reviewer: ```suggestion
* image:view-selector-icon.png[A table icon with several rows and columns] / image:view-selector-icon-2.png["A logo with a gold plus sign"] Toggle between the mapping view and the DataMapper configuration view.
```

> * image:save-icon.png["A gray floppy disk icon"]  Save the DataMapper properties

Reviewer: ```suggestion
* image:save-icon.png["A floppy disk icon"]  Save the DataMapper properties
```

|4 |
Input *Properties* tool. Click to open the *Properties* configuration window. Configurable properties will vary according to the input type. The image below shows the *Properties* configuration window.

image::csv-input-props.png["A dialog box for editing CSV format properties, including encoding, delimiters, and quote characters"]

|5 |

> *Show/Hide Unrelated Elements* tool. Default position is open, showing the hide icon image:hide-icon.png["A red line crossing out an eye icon"].

Reviewer: ```suggestion
*Show/Hide Unrelated Elements* tool. Default position is open, showing the hide icon image:hide-icon.png["A line crossing out an eye icon"].
```

Output *Properties* tool. Click to open the *Properties* configuration window. Configurable properties will vary according to the input type. Shown here are

|16 |

> *Show/Hide Unrelated Elements* tool. Default position is open, showing the hide icon image:hide-icon.png["A red line crossing out an eye icon"].

Reviewer: ```suggestion
*Show/Hide Unrelated Elements* tool. Default position is open, showing the hide icon image:hide-icon.png["A line crossing out an eye icon"].
```

DataMapper's preview feature allows you to test your mapping using a specific input file and preview the results. For details, see xref:previewing-datamapper-results-on-sample-data.adoc[Previewing DataMapper Results on Sample Data].

image::preview.png["A button with the label "Live" and an icon of a computer monitor"]

[%header,cols="10a,90a"]
|===
|Number |Description

> |1 |*Input Data* input field. Use to select the path of the file to read data from. Alternatively, click the folder image:input-folder-in-preview.png["A yellow folder with a green arrow"]button to the right of the field to select the file using your system's file browser.

Reviewer: ```suggestion
|1 |*Input Data* input field. Use to select the path of the file to read data from. Alternatively, click the folder image:input-folder-in-preview.png["A folder with a arrow"]button to the right of the field to select the file using your system's file browser.
```

== The DataMapper Configuration View

> The image:view-selector-icon.png["A table icon with several rows and columns"] / image:view-selector-icon-2.png["A green logo with a gold plus sign"] buttons to the right of the graphical mapping editor allow you to toggle between the mapping view and the DataMapper configuration view. The configuration view allows you to quickly configure global parameters for your specific DataMapper transformer.

Reviewer: ```suggestion
The image:view-selector-icon.png["A table icon with several rows and columns"] / image:view-selector-icon-2.png["A logo with a plus sign"] buttons to the right of the graphical mapping editor allow you to toggle between the mapping view and the DataMapper configuration view. The configuration view allows you to quickly configure global parameters for your specific DataMapper transformer.
```
